pkg/http-handler/http-handler_test.go

```diff
@@ -253,15 +254,18 @@ func TestValidateAppRepository(t *testing.T) {
 		name         string
 		err          error
 		expectedCode int
+		expectedBody string
 	}{
 		{
 			name:         "it should return OK if no error is detected",
 			expectedCode: 200,
+			expectedBody: "OK",
 		},
 		{
 			name:         "it should return the error code if given",
 			err:          fmt.Errorf("Boom"),
-			expectedCode: 500,
+			expectedCode: 200,
+			expectedBody: "Boom",
```

Comment:
Right, this no longer returns the error code. Sorry if it wasn't clear on the previous MP, but what I'd expect here is that *our* API call returns a 200 (as you've done), because it has done its job successfully, but the response data should contain the result of the validation. So here I'd expect that the (json) response include both the status code of the external request, as well as the error message (which you've included as the body).

So a successful validation returns a 200 with a body:
```
{
  valid: true
}
```

and for a validation that didn't succeed:
```
{
  valid: false,
  error: {
    code: 500,
    message: "Boom"
  }
}
```

or something similar. Yes, it works if we depend on the body text being the exact string "OK", but it limits what info we can give the user (ie. with the current change, a blank response from the external URL would give no info, whereas including both the externally received status code with the message will be clearer). But even without that, I'd be keen to avoid an API which requires the client to parse a string response to choose how to handle it.

Reply:
it's true that we are missing information without the status code